src/realm/query_conditions.hpp

```diff
@@ -65,13 +62,46 @@ class QueryStateBase {
     {
     }
 
+    virtual bool match(size_t, Mixed)
```

Comment:
Second note: I found it really confusing that `match` returns `false` on match, because it means "continue search" rather than "match found". I would love to have a comment about this in the interface definition.

Reply:
Suggestions followed